### DISCUSSION TITLE
Hide roles for users without permissions to read Directus Roles

### DISCUSSION THREAD

**Discussion by kropsi** [directus/directus#11202] [Archive]
>As "not admin" user you are still able to see all the roles in User Directory, also the ones that you are not assigned to. This must be a bug.

<details>
<summary>Original Post</summary>

> ### Preflight Checklist
> 
> - [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
> - [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
> - [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.
> 
> ### Describe the Bug
> 
> As "not admin" user you are still able to see all the roles in User Directory, also the ones that you are not assigned to. This must be a bug.
> 
> ### To Reproduce
> 
> 1. Create a user role with minimal access
> 2. Go to User Directory
> 
> ### Errors Shown
> 
> No errors, just a logical one
> 
> ### What version of Directus are you using?
> 
> 9.4.3.
> 
> ### What version of Node.js are you using?
> 
> not relevant
> 
> ### What database are you using?
> 
> not relevant
> 
> ### What browser are you using?
> 
> not relevant
> 
> ### What operating system are you using?
> 
> not relevant
> 
> ### How are you deploying Directus?
> 
> not relevant
> 

</details>

**Comment by joselcvarela** [comment]
I am moving this into Discussions since I consider this more a feature request than a bug.

**Comment by rijkvanzanten** [comment]
It's not a bug, it's based on the permissions for the role 👍🏻

By default, roles created with `App Access` get created with a set of _recommended_ permissions. You can still alter these and make them more strict. There's a button to hard-reset the permissions to the bare-minimums required for the app to function:

![CleanShot 2022-01-21 at 09 34 12@2x](https://user-images.githubusercontent.com/9141017/150544797-cd5b3918-6331-460a-984d-d18771fcddfe.png)
